api/internal/localizer/errors.go

```diff
@@ -16,3 +16,15 @@ func (rle ResourceLoadError) Error() string {
 	return fmt.Sprintf(`when parsing as inline received error: %s
 when parsing as filepath received error: %s`, rle.InlineError, rle.FileError)
 }
+
+type PathLocalizeError struct {
+	FileError error
+	RootError error
+}
+
+var _ error = PathLocalizeError{}
```

Comment:
super nit. remove this line

Reply:
@natasha41575 recommended this for readability: https://github.com/kubernetes-sigs/kustomize/pull/4869#discussion_r1033740486

I'm down either way.

Reply:
I think for errors, lines like this are not necessary.

Reply:
Yeah, in general the `var _ InterfaceX = StructX{}` is helpful since golang does not require specifying the implemented interfaces, so this static check makes the code author's purpose more clear. 
Though the `Error` interface is much less likely to change and need this sanity check. It's more like a convention.

Reply:
Done.